File: CSharp1/ConsoleInputOutput/QuadraticEquation/QuadraticEquation.cs
/* Problem 6: Write a program that reads the coefficients a, b and c of a quadratic equation ax2 + bx + c = 0
  and solves it (prints its real roots). */

using System;

class QuadraticEquation
{
    static void Main()
    {
        Console.Write("Enter a: ");
        double a = double.Parse(Console.ReadLine());
        Console.Write("Enter b: ");
        double b = double.Parse(Console.ReadLine());
        Console.Write("Enter c: ");
        double c = double.Parse(Console.ReadLine());

        double D = Math.Pow(b, 2) - 4 * a * c; //  Quadratic Formula -> D = b^2 - 4ac

        if (D < 0)      // This piece of code checks whether the equation has real roots, whether it has one or two
        {
            Console.WriteLine("No real roots!");
        }
        else if (D == 0)
        {
            double x = -b / (2 * a);
            Console.WriteLine("x1 = x2 = {0}", x);
        }
        else
        {
            double x1 = (-b + Math.Sqrt(D)) / (2 * a);
            double x2 = (-b - Math.Sqrt(D)) / (2 * a);

            Console.WriteLine("x1 = {0}\nx2 = {1}", x1, x2);

        }
    }

}

File: JavaScript/JavaScript-Fundamentals/ConditionalStatementsJS/ConditionalStatementsJS/QuadraticEquation.js
function solve() {

    alert('Problem 6: Quadratic equation');

    var a = prompt("a = "),
        b = prompt("b = "),
        c = prompt("c = "),
        sqrtpart = (b * b) - 4 * (a * c),
        x1,
        x2;

    if (sqrtpart > 0) {
        x1 = (-b + Math.sqrt(sqrtpart)) / (2 * a);
        x2 = (-b - Math.sqrt(sqrtpart)) / (2 * a);
    } else if (sqrtpart < 0) {
        x1 = 'no real root';
        x2 = 'no real root';
    } else {
        x1 = (-b + Math.sqrt(sqrtpart)) / (2 * a);
        x2 = x1;
    }

    alert('x1 = ' + x1 + ', x2 = ' + x2);

}